generator/internal/codegen_utils.cc

```diff
@@ -32,6 +33,81 @@ std::vector<std::pair<std::string, std::string>> const& SnakeCaseExceptions() {
       {"big_query", "bigquery"}};
   return kExceptions;
 }
+
+Status ProcessArgProductPath(
+    std::vector<std::pair<std::string, std::string>>& command_line_args) {
+  auto product_path =
+      std::find_if(command_line_args.begin(), command_line_args.end(),
+                   [](std::pair<std::string, std::string> const& p) {
+                     return p.first == "product_path";
+                   });
+  if (product_path == command_line_args.end() || product_path->second.empty()) {
+    return Status(StatusCode::kInvalidArgument,
+                  "--cpp_codegen_opt=product_path=<path> must be specified.");
+  }
+
+  auto& path = product_path->second;
+  if (path.front() == '/') {
+    path = path.substr(1);
+  }
+  if (path.back() != '/') {
+    path += '/';
+  }
+  return {};
+}
+
+Status ProcessArgGoogleapisCommitHash(
+    std::vector<std::pair<std::string, std::string>>& command_line_args) {
+  auto googleapis_commit_hash =
+      std::find_if(command_line_args.begin(), command_line_args.end(),
+                   [](std::pair<std::string, std::string> const& p) {
+                     return p.first == "googleapis_commit_hash";
+                   });
+  if (googleapis_commit_hash == command_line_args.end() ||
+      googleapis_commit_hash->second.empty()) {
+    return Status(
+        StatusCode::kInvalidArgument,
+        "--cpp_codegen_opt=googleapis_commit_hash=<hash> must be specified.");
+  }
+  return {};
+}
+
+void ProcessArgCopyrightYear(
+    std::vector<std::pair<std::string, std::string>>& command_line_args) {
+  auto copyright_year =
+      std::find_if(command_line_args.begin(), command_line_args.end(),
+                   [](std::pair<std::string, std::string> const& p) {
+                     return p.first == "copyright_year";
+                   });
+  if (copyright_year == command_line_args.end()) {
+    command_line_args.emplace_back("copyright_year", CurrentCopyrightYear());
+  } else if (copyright_year->second.empty()) {
+    copyright_year->second = CurrentCopyrightYear();
+  }
+}
+
+void ProcessArgOmitRpc(
+    std::vector<std::pair<std::string, std::string>>& command_line_args) {
+  absl::flat_hash_set<std::string> omitted_rpcs;
+  auto iter = std::find_if(command_line_args.begin(), command_line_args.end(),
+                           [](std::pair<std::string, std::string> const& p) {
+                             return p.first == "omit_rpc";
+                           });
+  while (iter != command_line_args.end()) {
+    omitted_rpcs.insert(iter->second);
+    command_line_args.erase(iter);
+    iter = std::find_if(command_line_args.begin(), command_line_args.end(),
+                        [](std::pair<std::string, std::string> const& p) {
+                          return p.first == "omit_rpc";
+                        });
+  }
+  if (!omitted_rpcs.empty()) {
```

Comment:
IIUC, this code is basically replacing a potential sequence of `omit_rpc=foo` flags with a single `omitted_rpcs=foo1,foo2` flag. If that's correct, can we avoid this by just requiring that the caller provide the `ommitted_rpcs` flag in the first place?